### PR TITLE
Create invoice

### PR DESCRIPTION
**Pull request by jrahaim**
Added commands for CreatBatchInvoice (non-standard) and cancel Batch invoice. 

Refactored some of the Dictionary -> Collection+JSON code.

Simmering while i test it against both my branch, develop, and any open invoicing branches.

### CODE REVIEW

File: TeamSnapSDK/SDK/DataLayer/TSDKDataRequest.m [hunk 1]
```diff
@@ -183,7 +183,9 @@ + (void)requestJSONObjectsForPath:(NSURL *)URL sendDataDictionary:(NSDictionary
             [request setHTTPBody:data];
             [request setValue:@"application/json; charset=utf-8" forHTTPHeaderField:@"Content-Type"];
         }
-
+        
+#define DEBUGCURL
```
Comment (larsacus) [inline TeamSnapSDK/SDK/DataLayer/TSDKDataRequest.m]: Was this meant to be checked in?

Comment (jrahaim) [reply TeamSnapSDK/SDK/DataLayer/TSDKDataRequest.m]: Nope.

File: TeamSnapSDK/SDK/DataTypes/TSDKBatchInvoice.m [hunk 2]
```diff
@@ -29,4 +44,87 @@ - (TSDKInvoiceStatus)invoiceStatus {
     }
 }
 
++ (void)createInvoicesWithDueDate:(NSDate *_Nonnull)dueDate teamId:(NSString *_Nonnull)teamId title:(NSString *_Nonnull)title description:(NSString *_Nullable)description invoiceType:(TSDKInvoiceCategory)invoiceType invoiceLineItems:(NSArray *_Nonnull)invoiceLineItems members:(NSArray<TSDKMember *> *)members isRecipientPayingTransactionFees:(BOOL)isRecipientPayingTransactionFees completion:(TSDKBatchInvoiceCreatedBlock _Nullable)completion {
+    
+    TSDKCollectionCommand *createInvoiceCommand = [[self commandForKey:@"create_with_invoices"] copy];
+    
+    // Hacky hack: Do not merge
+    if ([createInvoiceCommand.href containsString:@"create_with_invoices"] == NO) {
```
Comment (larsacus) [inline TeamSnapSDK/SDK/DataTypes/TSDKBatchInvoice.m]: Teamsnap doublecheck for this hacky hack

File: TeamSnapSDK/SDK/DataTypes/TSDKBatchInvoice.m [hunk 3]
```diff
@@ -29,4 +44,87 @@ - (TSDKInvoiceStatus)invoiceStatus {
...
+        }
+        createInvoiceCommand.data[@"member_ids"] = membersStrings;
+        [createInvoiceCommand executeCollectionJSONTemplateWithCompletion:^(BOOL success, BOOL complete, TSDKCollectionJSON * _Nullable objects, NSError * _Nullable error) {
+            
+            TSDKBatchInvoice *batchInvoice = nil;
+            NSMutableArray <TSDKBatchInvoiceLineItem *> *batchInvoiceLineItems = [[NSMutableArray alloc] init];
+            NSMutableArray <TSDKInvoice *> *invoices = [[NSMutableArray alloc] init];
+            
+            if (success && ([[objects collection] isKindOfClass:[NSArray class]])) {
+                // This currently returns a mixed bag of batch_invoice, batch_invoice_line
```
Comment (larsacus) [inline TeamSnapSDK/SDK/DataTypes/TSDKBatchInvoice.m:87]: wow

**Comment by jrahaim** [comment]
All issues have been addressed.

**Comment by larsacus** [comment]
This looks good to me.

**Pull request merged**